File: python/solutions/leetcode/easy/repeated_substring_pattern.py
# https://leetcode.com/problems/repeated-substring-pattern 459. Repeated Substring Pattern

class RepeatedSubstringPattern:
    def repeatedSubstringPattern(self, s: str) -> bool:
        n = len(s)
        k = 1

        while k < n:
            substring = s[:k]
            if substring * (n // k) == s:
                return True
            k += 1

        return False

File: java/src/main/java/org/solutions/leetcode/easy/RepeatedSubstringPattern.java
package org.solutions.leetcode.easy;

/**
 * <a href="https://leetcode.com/problems/repeated-substring-pattern">459. Repeated Substring Pattern</a>
 */

public class RepeatedSubstringPattern {
    public boolean repeatedSubstringPattern(String s) {
        return (s + s).substring(1, 2 * s.length() - 1).contains(s);
    }
}
